common/http_timing.py

```
@@ -0,0 +1,222 @@
+            except Exception as e:
+                last_exception = e
+                if retry_count < MAX_RETRIES - 1:
+                    # Exponential backoff for connection errors
+                    await asyncio.sleep(2 ** retry_count)
+                    continue
+                raise
+
+        if not response:
+            if last_exception:
+                raise last_exception
+            raise ValueError("No response received after retries")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)